src/components/shared/LanguageSelector/LanguageSelector.test.js

```diff
@@ -62,5 +81,29 @@ describe('Language Selector Component', () => {
     );
     // Assert
     expect(getByText('ES')).toBeInTheDocument();
+    expect(container.querySelector('.lang--es')).toBeInTheDocument();
+    expect(container.querySelector('.lang--es').search).toEqual(
+      '?lang=es&test&redirect=test&test2',
+    );
+  });
+
+  it('should show ES and maintain all parameters when lang parameter is in the end', () => {
```

Comment:
How about adding a scenario for lang in the start of search, and lang in the middle. I' ve seen you tested this manually :trollface:

Reply:
haha i added now!